modules/xfeatures2d/src/beblid.cpp

```diff
@@ -0,0 +1,602 @@
+ */
+inline void rectifyABWL(const std::vector<ABWLParams> &wlPatchParams,
+                        std::vector<ABWLParams> &wlImageParams,
+                        const cv::KeyPoint &kp,
+                        float scaleFactor = 1,
+                        const cv::Size &patchSize = cv::Size(32, 32))
+{
+    float m00, m01, m02, m10, m11, m12;
+    float s, cosine, sine;
+    uint32_t i;
+
+    s = scaleFactor * kp.size / (0.5f * (patchSize.width + patchSize.height));
+    wlImageParams.resize(wlPatchParams.size());
+
+    if (kp.angle == -1)
+    {
+        m00 = s;
+        m01 = 0.0f;
+        m02 = -0.5f * s * patchSize.width + kp.pt.x;
+        m10 = 0.0f;
+        m11 = s;
+        m12 = -s * 0.5f * patchSize.height + kp.pt.y;
+    }
+    else
+    {
+        cosine = (kp.angle >= 0) ? float(cos(kp.angle * CV_DEGREES_TO_RADS)) : 1.f;
+        sine = (kp.angle >= 0) ? float(sin(kp.angle * CV_DEGREES_TO_RADS)) : 0.f;
+
+        m00 = s * cosine;
+        m01 = -s * sine;
+        m02 = (-s * cosine + s * sine) * patchSize.width * 0.5f + kp.pt.x;
+        m10 = s * sine;
+        m11 = s * cosine;
+        m12 = (-s * sine - s * cosine) * patchSize.height * 0.5f + kp.pt.y;
+    }
+
+    for (i = 0; i < wlPatchParams.size(); i++)
+    {
+        wlImageParams[i].x1 = CV_ROUNDNUM(m00 * wlPatchParams[i].x1 + m01 * wlPatchParams[i].y1 + m02);
+        wlImageParams[i].y1 = CV_ROUNDNUM(m10 * wlPatchParams[i].x1 + m11 * wlPatchParams[i].y1 + m12);
+        wlImageParams[i].x2 = CV_ROUNDNUM(m00 * wlPatchParams[i].x2 + m01 * wlPatchParams[i].y2 + m02);
+        wlImageParams[i].y2 = CV_ROUNDNUM(m10 * wlPatchParams[i].x2 + m11 * wlPatchParams[i].y2 + m12);
+        wlImageParams[i].boxRadius = CV_ROUNDNUM(s * wlPatchParams[i].boxRadius);
+    }
+}
+
+/**
+ * @brief Computes the Average Box Weak-Learner response, measuring the difference of
+ * gray level in the two square regions.
+ * @param wlImageParams The weak-learner parameter defining the size and locations of each box.
+ * @param integralImage The integral image used to compute the average gray value in the square regions.
+ * @return The difference of gray level in the two squares defined by wlImageParams
+ */
+inline float computeABWLResponse(const ABWLParams &wlImageParams, const cv::Mat &integralImage)
+{
+    CV_DbgAssert(!integralImage.empty());
+
+    int frameWidth, frameHeight, box1x1, box1y1, box1x2, box1y2, box2x1, box2y1, box2x2, box2y2;
+    int idx1, idx2, idx3, idx4;
+    int A, B, C, D;
+    const int *ptr;
+    int box_area1, box_area2, width;
+    float sum1, sum2, average1, average2;
+    // Since the integral image has one extra row and col, calculate the patch dimensions
+    frameWidth = integralImage.cols;
+    frameHeight = integralImage.rows;
+
+    // For the first box, we calculate its margin coordinates
+    box1x1 = wlImageParams.x1 - wlImageParams.boxRadius;
+    if (box1x1 < 0) box1x1 = 0;
+    else if (box1x1 >= frameWidth - 1) box1x1 = frameWidth - 2;
```

Comment:
dedicated lines for `if` body (code coverage).

Reply:
done, I have fixed this problem also other lines where the body was in the same line. Thank you